Mage.Sets/src/mage/cards/a/AnzragsRampage.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    private static final ValueHint hint = new ValueHint(
+            "Artifacts put into graveyards from the battlefield this turn", AnzragsRampageValue.instance);
+
+    public AnzragsRampage(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{R}{R}");
+
+        // Destroy all artifacts you don't control, then exile the top X cards of your library,
+        // where X is the number of artifacts that were put into graveyards from the battlefield this turn.
+        // You may put a creature card exiled this way onto the battlefield. It gains haste.
+        // Return it to your hand at the beginning of the next end step.
+        this.getSpellAbility().addEffect(new DestroyAllEffect(filter));
+        this.getSpellAbility().addEffect(new AnzragsRampageEffect().concatBy(", then"));
+        this.getSpellAbility().addWatcher(new AnzragsRampageWatcher());
+        this.getSpellAbility().addHint(hint);
+    }
+
+    private AnzragsRampage(final AnzragsRampage card) {
+        super(card);
+    }
+
+    @Override
+    public AnzragsRampage copy() {
+        return new AnzragsRampage(this);
+    }
+}
+
+// CardsPutIntoGraveyardWatcher does not count tokens so custom watcher is needed.
+// Copied from StructuralAssaultWatcher
+class AnzragsRampageWatcher extends Watcher {
+
+    private int artifactsDied;
+
+    AnzragsRampageWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.ZONE_CHANGE) {
+            ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
+            if (zEvent.isDiesEvent() && zEvent.getTarget().isArtifact(game)) {
+                artifactsDied++;
+            }
+        }
+    }
+
+    @Override
+    public void reset() {
+        super.reset();
+        artifactsDied = 0;
+    }
+
+    public int getArtifactsDied() {
+        return artifactsDied;
+    }
+}
+
+enum AnzragsRampageValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        AnzragsRampageWatcher watcher = game.getState().getWatcher(AnzragsRampageWatcher.class);
+        if (watcher == null) {
+            return 0;
+        }
+        return watcher.getArtifactsDied();
+    }
+
+    @Override
+    public AnzragsRampageValue copy() {
+        return instance;
+    }
+
+    @Override
+    public String getMessage() {
+        return "";
+    }
+}
+
+class AnzragsRampageEffect extends OneShotEffect {
+
+    AnzragsRampageEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "exile the top X cards of your library, where X is the number of artifacts " +
+                "that were put into graveyards from the battlefield this turn. " +
+                "You may put a creature card exiled this way onto the battlefield. It gains haste. " +
```

Comment:
Its’s a big effect with multiple steps/actions, so add related card rules in the code blocks as comments for easier review and modify.